1 table.sql

```diff
@@ -0,0 +1,38 @@
+use CostumeDB
+go
+
+
+drop table if exists dbo.Costume
+go 
+create table dbo.Costume(
+    CostumeId int not null identity primary key,
+    CustomerFirstName varchar(30) not null
+        constraint c_Costume_Customer_First_Name_cannot_be_blank check(CustomerFirstName <> ''),
+    CustomerLastName varchar(30) not null
+        constraint c_Costume_Customer_Last_Name_cannot_be_blank check(CustomerLastName <> ''),
+    Costume varchar(30) not null 
+        constraint c_Costume_Costume_can_be_AmericanGirlDoll_Artist_BumbleBee_ColonialBoy_ColonialGirl_Elephant_FireMan_PoliceMan_Princess_or_Zebra
+         check(Costume in ('American Girl Doll', 'Artist', 'Bumble Bee', 'Colonial Boy', 'Colonial Girl', 'Elephant', 'Fire Man', 'Police Man', 'Princess', 'Zebra')),
+    Size varchar(2) not null
+        constraint c_Costume_Size_can_be_XS_S_M_L_or_XL check(Size in ('XS', 'S', 'M', 'L', 'XL')),
+    AmountBought int not null 
+        constraint c_Costume_Amount_bought_must_be_1_or_more check(AmountBought >= 1),
+    SoldPricePerCostume int not null, 
+    DateBought date not null,
+    DateSold date not null
+        constraint c_Costume_Date_Sold_can_be_after_January_1_2020_and_cannot_be_future_date check(DateSold between 'January 1 2020' and getdate()),
+    BoughtPricePerCostume as case 
+                                when Size = 'XS' then 15
+                                when Size = 'S' then 17
+                                when Size = 'M' then 20
+                                when Size = 'L' then 22
+                                when Size = 'XL' then 25
+                            end persisted,
+    TotalPaid as AmountBought*SoldPricePerCostume persisted,
+    constraint c_Costume_Sold_Price_must_be_greater_than_or_equal_to_Bought_Price_per_costume check(SoldPricePerCostume >= BoughtPricePerCostume),
+    constraint c_Costume_Date_Bought_must_be_before_or_same_as_Date_Sold check(DateBought <= DateSold) 
+    
+)
```

Comment:
LB: You should also record if the customer paid full price. Add a column for that. (the column can be computed on cost price and price sold...)

Reply:
I added it. I couldn't do it based on cost price because it's already a computed column....

Reply:
Not sure what you mean... can you please be more specific? Thank you!

Reply:
Sorry, I meant to say that you can check if sold price is same as total paid but realizing now that TotalPaid is a computed column so that wouldn't work. Keep it the way it is but do change it to datatype bit 0/1 instead of yes/no.